glommio/src/executor/mod.rs

```diff
@@ -1826,14 +1826,19 @@ impl ExecutorProxy {
     where
         F: FnOnce(&LocalExecutor) -> bool,
     {
-        let need_yield = LOCAL_EX.with(|local_ex| {
-            if cond(local_ex) {
-                local_ex.mark_me_for_yield();
-                true
-            } else {
-                false
-            }
-        });
+        let need_yield = if LOCAL_EX.is_set() {
+            LOCAL_EX.with(|local_ex| {
+                if cond(local_ex) {
+                    local_ex.mark_me_for_yield();
+                    true
+                } else {
+                    false
+                }
+            })
+        } else {
+            // We are not in a glommio context
+            true
```

Comment:
that will kill performance, no?
If you returned from this function right now it would not be that bad, but right below this line we use this result to `yield_now()`

Reply:
Good point